app/frontends/stac-catalog.ts

```diff
@@ -88,29 +86,35 @@ class HarmonyCatalog implements SerializableCatalog {
 /**
  * Function to create the STAC Catalog given a Harmony Job object
  *
- * @param job - Harmony Job object
+ * @param jobID - Harmony job jobID string
+ * @param jobRequest - Harmony job, job request string
+ * @param stacDataLinks - JobLink's to add to catalog
+ * @param pagingLinks - links pointing to the next, previous, and current page
  * @param linkType - the type of data links that the stac-items should use
  *
  * @returns - STAC Catalog JSON
  *
  * @example
  * const catalog = require('HarmonyCatalog');
- * let jsonObj = catalog.create(job);
+ * let jsonObj = catalog.create(job.jobID, job.request, stacDataLinks, pagingLinks);
```

Comment:
I can't remember perfectly because there were a number of constraints I was thinking about early on (that I thought this satisfied well), but I'll try my best to explain, first by explaining a separate but similar change.

stacItem.create was previously indexing into an array of job.links based on the index requested by the user and generating the STAC item based on that link. Since we're no longer loading all job links into memory, it's no longer safe to assume in stacItem.create that job.links[index] has the link that we want. So that change decouples the structure/ordering of job.links from the catalog.create logic.

Once I went down that road, it made sense for consistency to have catalog.create also force the caller to specify up front which links they want turned into STAC items/links rather than relying directly on a job. 

Happy to revert if you disagree. I could definitely see how the simplicity of the former might be cleaner.